src/auth/strategies/util.ts

```diff
@@ -25,37 +25,45 @@ export async function ExternalServiceCallback(
     // If `user` exists, the user has already logged in with this service and is good-to-go
     let user = await User.findOne({ [`services.${serviceName}.id`]: id });
 
-    if (session && session.email && session.firstName && session.lastName) {
-        let signupEmail = session.email.trim().toLowerCase();
+    let enteredEmail = session.email?.trim().toLowerCase();
+
+    // Ensure that the oauth method used is linked to the user with the same email address that is entered
+    if (enteredEmail && user && enteredEmail !== user.email) {
+        done(null, false, { message: "The 3rd party account you chose does not match the email address that you are trying to log in with. Please make sure your email address and the selected account match." });
```

Comment:
Can you clarify the exact expectation in the error message by adding what aspect of the third-party account doesn't match (you mention the email address the user entered but the rest is vague)?

Reply:
Basically, I want them to login with a different 3rd party account. So Ill just tell them that. Let me know if this is good instead.

The {serviceName} account you selected is linked to a different email address than typed below. Please try again with a different 3rd party account.

Where service name is google, github, etc.